armi/reactor/components/complexShapes.py

```diff
@@ -140,6 +137,69 @@ def getCircleInnerDiameter(self, Tc=None, cold=False):
         return self.getDimension("holeOP", Tc, cold)
 
 
+class FilletedHexagon(basicShapes.Hexagon):
+    """Hexagon with rounded corners.
+
+    This shape has a hexagonal hole with rounded corners cut out of the center of it. By default,
+    that inner hole has a diameter of zero, making this a solid object with no hole.
+    """
+
+    THERMAL_EXPANSION_DIMS = {"cornerR", "ip", "op"}
+
+    pDefs = componentParameters.getFilletedHexagonParameterDefinitions()
+
+    def __init__(
+        self,
+        name,
+        material,
+        Tinput,
+        Thot,
+        op,
+        ip=0.0,
+        cornerR=0.0,
+        mult=1.0,
+        modArea=None,
+        isotopics=None,
+        mergeWith=None,
+        components=None,
+    ):
+        ShapedComponent.__init__(
+            self,
+            name,
+            material,
+            Tinput,
+            Thot,
+            isotopics=isotopics,
+            mergeWith=mergeWith,
+            components=components,
+        )
+        self._linkAndStoreDimensions(
+            components, op=op, ip=ip, cornerR=cornerR, mult=mult, modArea=modArea
+        )
+        assert cornerR <= op / 2, "The radius of curvature is too large."
+
+    @staticmethod
+    def _area(D, r):
+        """Helper function, to calculate the area of a hexagon with rounded corners."""
+        if D <= 0.0:
+            return 0.0
+
+        area = 1.0 - (1.0 - (math.pi / (2.0 * math.sqrt(3)))) * (2 * r / D) ** 2
+        area *= (math.sqrt(3.0) / 2.0) * D**2
+        return area
+
+    def getComponentArea(self, cold=False, Tc=None):
+        """Computes the area for the rounded hexagon component in cm^2."""
+        op = self.getDimension("op", cold=cold, Tc=None)
+        ip = self.getDimension("ip", cold=cold, Tc=None)
+        r = self.getDimension("cornerR", cold=cold, Tc=None)
```

Comment:
Why are these three calls not passing through the value of `Tc`?

Reply:
typo! Fixing!